subprojects/docs/src/docs/userguide/troubleshooting/validation_problems.adoc

```diff
@@ -57,3 +57,117 @@ To fix this problem, you must remove the property and use the `TransformOutputs`
 
+[[ignored_annotations_on_field]]
+== Invalid use of annotations on fields
+
+This error indicates that you have a field annotated, but that there's no getter for this field.
+Gradle will recognize annotations on fields _only_ if there's a corresponding getter.
+If this getter is absent, the annotations have no effect whatsoever.
+
+To fix this, you need to create a getter for this field.
+We also recommend to annotate the getter instead of the field.
+
+If you are using Groovy, it's likely that you unintentionally added the `private` modifier to a property declaration:
+
+```groovy
+@InputFile
+RegularFileProperty inputFile // this is a public property
+
+@InputFile
+private RegularFileProperty inputFile // this is a field, remove the `private` modifier
+```
+
+[[ignored_annotations_on_method]]
+== Invalid annotation on method
+
+This error indicates that a annotation has been placed on an unexpected method.
+In general, annotations like `@InputFiles` or `@OutputDirectory` need to be placed on a "property" method.
+A property is defined by having a getter.
+
+This causes Gradle to ignore the annotations, so they are typically not used in up-to-date checking or caching.
+
+To fix this problem, you must remove the annotation, create a getter for the property you want to use and annotate the getter instead.
+
+[[mutable_type_with_setter]]
+== Mutable type with setter
+
+This error indicates that a property of "mutable" type is also providing a setter.
+Mutable types in Gradle include link:{javadocPath}/org/gradle/api/provider/Property.html[Property] or link:{javadocPath}/org/gradle/api/file/ConfigurableFileCollection.html[ConfigurableFileCollection].
+
+For example, you wrote:
+
+```groovy
+class MyTask extends DefaultTask {
+    private Property<Integer> x
+
+    @Input
+    Property<Integer> getX() { this.x }
+
+    void setX(Property<Integer> x) { this.x = x }
+}
+```
+
+However, mutable types like `Property` are designed to track dependencies and origin of values they are supplied with.
+As a consequence, it is an error to make them overwritable, because Gradle will then be unable to tell where a property value comes from.
+
+To fix this problem, you should either make your property final and remove the setter:
+
+```groovy
+class MyTask extends DefaultTask {
+    @Input
+    final Property<Integer> getX() = ...
+```
+
+or rely on Gradle's built-in capacity to inject final properties:
+
+```groovy
+abstract class MyTask {
+    abstract Property<Integer> getX()
+}
+```
+
+then the value of the properties need to be configured via the mutation methods:
+
+```groovy
+myTask {
+    x.set(123)
+}
+```
+
+[[redundant_getters]]
+== Redundant getters
+
+This error indicates that a `boolean` property has both a `get` and an `is` getter method.
+This is a problem because both getters can be annotated differently and Gradle cannot know which ones to use.
+
+The solution to this problem is to get rid of one of the getters.
```

Comment:
I think you can also annotate one of the getters with `@Internal`, though I am not 100% sure. Do we have tests for doing so?

Reply:
I don't know.

Reply:
Looks like we do: https://github.com/gradle/gradle/blob/451b05c9a65353f519af77a6b85bb978bc5b5545/subprojects/model-core/src/test/groovy/org/gradle/internal/reflect/annotations/impl/DefaultTypeAnnotationMetadataStoreTest.groovy#L222-L228

Can you add this as a solution here?

Reply:
I will!